src/runtime_src/xdp/profile/plugin/aie_trace/aie_trace_plugin.cpp

```diff
@@ -307,12 +311,19 @@ void AieTracePluginUnified::updateAIEDevice(void *handle, bool hw_context_flow)
       ,
       AIEData.metadata->getNumStreams(), devInst);
 #else
+  XAie_DevInst* devInst = aie::trace::getAieDevInst(handle, hw_context_flow);
```

Comment:
Can we have a < TraceImpl >::"setAieDeviceInst" for which will call the "fetchAieDevInst" for its backend and populate VPStaticDb::aieDevInst ? For now, we can have it implemented for VE2 and Edge. This < TraceImpl >::"setAieDeviceInst" can be called before instantiating trace offloader, so that trace offloader can directly use the AieDevInst in StaticDB.